### ISSUE TITLE
Ability to show a spinner to indicate progress when saving data on server side

### ISSUE THREAD

**Issue by sushil-rxr** [vitalets/angular-xeditable#478]
Hi,
I am using the xeditable in-place edit quite frequently and one use case is where I want to save data to the server in a beforesave method. While the data is getting saved and I get back the response, I want to be able to show a progress indicator (through a spinner maybe) within the text box.

I am using the Bootstrap theme and it does not allow me to add feedback classes which I can use to get the desired behaviour. However, I am not sure how to get this template working with xeditable as I dont have control over the HTML templates used to render the editable content. 

For example, to get my desired outcome, I want to be able to have this, 

<div class="form-group has-success has-feedback">
  <label class="control-label">Some field</label>
  <input type="text" class="form-control">
  <span class="fa fa-spin fa-cog form-control-feedback" ng-if="condition"></span>
</div>

Is there any way we can have support for this out-of-the-box using angular-xeditable?

**Comment by ckosloski** [comment]
It seems like you forgot to paste some code in your question.

It is possible to override the templates if that is what you are looking for.  See [here](http://vitalets.github.io/angular-xeditable/#default)

**Comment by sushil-rxr** [comment]
Yeah, thanks for pointing out. I had the code but did not mark it as code, so it was not appearing :-)

Is it also possible to override the input templates using the approach you suggested? And is it possible to specify it for only particular instance rather than globally?

**Comment by ckosloski** [comment]
I'm not sure, I haven't tried overriding the templates.

**Comment by ckosloski** [comment]
@sushil-rxr You can change the theme per input (i.e. default or bs2/bs3,etc), but you can't change how the template looks per input, it is a global setting.
As for adding the spinner, looking at this again, it would require some code changes to add an optional spinner.  I don't see how else to easily add this to the existing template layout.
